tests/plugins/widget/editing.js

```diff
@@ -335,6 +360,24 @@
 			} );
 		},
 
+		// (#1044)
+		'test opening and hiding dialog events': function() {
+			var editor = this.editor;
+
+			this.editorBot.setData( '<p data-widget="dialogtest1" id="x">bar</p>', function() {
+				var widget = getWidgetById( editor, 'x' );
+
+				editor.once( 'dialogShow', function( evt ) {
+					evt.data.getButton( 'ok' ).click();
+					resume();
```

Comment:
Moving assertions outside the test harms its readability. At the first glance this test looks like always failing one as there is no assertion in it. Please move assertions into the test itself.